File: src/main/java/jp/dataforms/fw/field/common/RowNoField.java
package jp.dataforms.fw.field.common;

import jp.dataforms.fw.field.sqltype.IntegerField;

/**
 * 検索結果の行番号フィールドクラス。
 *
 */
public class RowNoField extends IntegerField {
	/**
	 * コンストラクタ。
	 */
	public RowNoField() {
		super(null);
		this.setComment("No.");
	}

	/**
	 * コンストラクタ。
	 * @param id フィールドID。
	 */
	public RowNoField(final String id) {
		super(id);
		this.setComment("No.");
	}
}

File: src/main/java/META-INF/resources/dataforms/field/common/RowNoField.js
/**
 * @fileOverview {@link RowNoField}クラスを記述したファイルです。
 */

'use strict';

import { IntegerField } from '../sqltype/IntegerField.js';

/**
 * @class RowNoField
 *
 * @extends IntegerField
 */
export class RowNoField extends IntegerField {
	/**
	 * コンストラクタ。
	 */
	constructor() {
		super();
	}

	/**
	 * HTMLエレメントとの対応付けを行います。
	 */
	attach() {
		super.attach();
	}
}

